docs/cartesian-product.adoc

```diff
@@ -292,21 +272,21 @@ To demonstrate with a table:
 |===
 
 For more information, please see the link:range-sources.adoc[separate documentation about range sources].
-You can combine range sources with `@CartesianValueSource`, it works as you'd expect:
+You can combine range sources with any other annotation supported by `@CartesianProductTest`, it works as you'd expect:
 
 [source, java]
 ----
 @CartesianProductTest
-@ShortRangeSource(from = 1, to = 3)
-@CartesianValueSource(strings = { "text one", "text two" })
-void mixAndMatchTest(short number, String text) {
+void mixAndMatchTest(
+		@ShortRangeSource(from = 1, to = 3) short number,
+		@CartesianValueSource(strings = { "text one", "text two" }) String text) {
 	// passing test code
 }
 ----
 
 == Writing a static factory method for the parameters
 
-If your tests require special inputs that `@CartesianValueSource` is not able to supply, you can define a static factory method to supply your test parameters.
+If your tests require special inputs that supported annotations are not able to supply, you can define a static factory method to supply your test parameters.
```

Comment:
At first I found it weird that the static factories don't move from "method level" to "parameter level" as well. The latter would mean having a `@FactorySource` (or something) on the method parameter that would then provide argument just for that one. It makes more sense to keep the current approach, though. Because it is incompatible with the parameter annotations, we should clarify that in the "what works where" portion as well.

That said, maybe it makes sense to implement `@FactorySource` _in addition_ to the existing capabilities? Probably in its own issue/PR.

Reply:
I think before we release 2.0, we should remove the `factory` attribute of `@CartesianProductTest` (and the implicit static factory based on test method name) and create the `@CartesianMethodSource` or something similar.

We should also consider doing the same thing for the `value` attribute - removing and adding a dedicated annotation instead.